File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Projects/readingtabledata.rb
module Asposetasksjava
  module ReadingTableData
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        t1 = project.getTables().toList().get(0)
        puts "Table Fields Count" + t1.getTableFields().size().to_s

        f = t1.getTableFields().get(0)
        puts "Field width: " + f.getWidth().to_s
        puts "Field Title: " + f.getTitle().to_s
        puts "Field Title Alignment: " + f.getAlignTitle().to_s
        puts "Field Align Data: " + f.getAlignData().to_s

        f = t1.getTableFields().get(1)
        puts "Field width: " + f.getWidth().to_s
        puts "Field Title: " + f.getTitle().to_s
        puts "Field Title Alignment: " + f.getAlignTitle().to_s
        puts "Field Align Data: " + f.getAlignData().to_s
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithProjects/ReadingTableData.php
<?php
namespace Aspose\Tasks\WorkingWithProjects;

use com\aspose\tasks\Project as Project;
class ReadingTableData{

    public static function run($dataDir=null){
        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $t1 = $project->getTables()->toList()->get(0);
        print "Table Fields Count" . (string)$t1->getTableFields()->size();//.to_s

        $f = $t1->getTableFields()->get(0);
        print "Field width: " . (string)$f->getWidth().PHP_EOL;//.to_s
        print "Field Title: " . (string)$f->getTitle().PHP_EOL;//.to_s
        print "Field Title Alignment: " . (string)$f->getAlignTitle().PHP_EOL;//.to_s
        print "Field Align Data: " . $f->getAlignData().PHP_EOL;//.to_s

        $f = $t1->getTableFields()->get(1);
        print "Field width: " . (string)$f->getWidth().PHP_EOL;//.to_s
        print "Field Title: " . (string)$f->getTitle().PHP_EOL;//.to_s
        print "Field Title Alignment: " . (string)$f->getAlignTitle().PHP_EOL;//.to_s
        print "Field Align Data: " . (string)$f->getAlignData().PHP_EOL;//.to_s
    }

}
?>